pkg/agent/svid/rotator.go

```diff
@@ -124,15 +135,79 @@ func (r *rotator) SetRotationFinishedHook(f func()) {
 }
 
 func (r *rotator) rotateSVIDIfNeeded(ctx context.Context) (err error) {
-	if rotationutil.ShouldRotateX509(r.clk.Now(), r.state.Value().(State).SVID[0]) {
-		err = r.rotateSVID(ctx)
+	state, ok := r.state.Value().(State)
+	if !ok {
+		return fmt.Errorf("unexpected value type: %T", r.state.Value())
 	}
-	if r.rotationFinishedHook != nil {
-		r.rotationFinishedHook()
+
+	if rotationutil.ShouldRotateX509(r.clk.Now(), state.SVID[0]) {
+		if state.Reattestable {
```

Comment:
After thinking about this, and due to the criticality of this particular flow, i think we should put reattestation-for-renewal behind a feature flag that we can default false for now. I know this will extend the deadline for forcing agents to reattest, but it should give us some more time to put this through its paces. See the pkg/common/fflag package.

Reply:
i think so too. this change is in.